logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveIsSelfConversationMemberUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.feature.user.GetSelfUserUseCase
+import com.wire.kalium.logic.functional.fold
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+
+class ObserveIsSelfConversationMemberUseCase(
```

Comment:
```suggestion
class ObserveIsSelfUserMemberUseCase(
```

Reply:
I think this use case should also be split in interface + implementation

Reply:
So that we can also **unit test** this use case

Reply:
Finally done 😌